File: day01/src/main/python/solution.py
def count_increments(sonar):
    count = 0
    for i in range(len(sonar) - 1):
        if sonar[i + 1] > sonar[i]:
            count = count + 1
    return count


f = open('../resources/input', 'r')
lines = list(map(int, f.readlines()))
print("part 1")
print(count_increments(lines))

grouped = [lines[i + 0] + lines[i + 1] + lines[i + 2] for i in range(len(lines) - 2)]
print("part 2")
print(count_increments(grouped))

File: day01/src/main/cpp/solution.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>

#define VLL vector<long long>

using namespace std;

VLL readInputFile() {
    ifstream file("../resources/input");
    VLL lines;
    string s;
    while(getline(file, s)) {
        lines.push_back(stoll(s));
    }
    return lines;
}

int countDeeperLines(VLL sonar) {
    int count = 0;
    for(int i = 0; i < sonar.size() - 1; i++){
        if(sonar[i+1] > sonar[i]) count++;
    }
    return count;
}

int main() {
    VLL sonar = readInputFile();
    cout << "part 1" << endl;
    cout << countDeeperLines(sonar) << endl;
    cout << "part 2" << endl;
    VLL windowed;
    for(int i=0;  i < sonar.size() - 2; i++) {
        windowed.push_back(sonar[i]+sonar[i+1]+sonar[i+2]);
    }
    cout << countDeeperLines(windowed) << endl;
    return 0;
}
